lib/off_broadway_twitter/producer.ex

```diff
@@ -3,98 +3,148 @@ defmodule OffBroadwayTwitter.Producer do
 
   alias Broadway.Producer
   alias Broadway.Message
-  alias OffBroadwayTwitter.TwitterFinch
+
+  alias Mint.HTTP2
 
   @behaviour Producer
   @twitter_stream_url_v2 "https://api.twitter.com/2/tweets/sample/stream"
+  @connect_in_ms 2_000
+  @reconnect_in_ms 1_000
 
   @impl true
   def init(opts) do
-    Process.flag(:trap_exit, true)
+    uri = URI.parse(@twitter_stream_url_v2)
+    token = Keyword.fetch!(opts, :twitter_bearer_token)
+
+    timer = schedule_connection(@connect_in_ms)
```

Comment:
Shouldn't the first connection happen immediately? :)

Reply:
In my mind I shouldn't do more than work with opts and the initial setup inside `init`. Good to know that is not the case o/